src/transformers/trainer.py

```diff
@@ -307,6 +309,8 @@ def train(self, model_path: Optional[str] = None):
 
         tr_loss = 0.0
         logging_loss = 0.0
+        best_eval_loss = None
+        evals_without_improvement = 0
```

Comment:
nit: prefix those with `patience_` as they're specific to this

```suggestion
        patience_best_eval_loss = None
        patience_evals_without_improvement = 0
        patience_should_stop = False
```